frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/LibraryOptionsEditor/LibraryOptionsEditor.tsx

```diff
@@ -25,11 +30,28 @@ export function LibraryOptionsEditor({
   const { t } = useTranslation();
   const { org, app } = useStudioEnvironmentParams();
   const { data: optionList } = useOptionListQuery(org, app, optionListId);
-  const { doReloadPreview } = usePreviewContext();
+  const { data: textResources } = useTextResourcesQuery(org, app);
   const { mutate: updateOptionList } = useUpdateOptionListMutation(org, app);
+  const { mutate: updateTextResource } = useUpsertTextResourceMutation(org, app);
+  const { doReloadPreview } = usePreviewContext();
   const editorTexts: CodeListEditorTexts = useOptionListEditorTexts();
   const modalRef = createRef<HTMLDialogElement>();
 
+  const textResourcesForLanguage = useMemo(
+    () => getTextResourcesForLanguage(language, textResources),
+    [textResources],
+  );
+
+  const handleBlurTextResource = useCallback(
+    (textResource: TextResource) => {
+      const updatedTextResource = createTextResourceWithLanguage(language, textResource);
+      const mutationArgs = convertTextResourceToMutationArgs(updatedTextResource);
+      updateTextResource(mutationArgs);
+      doReloadPreview();
+    },
+    [updateTextResource, doReloadPreview],
+  );
```

Comment:
What do you think about moving this into a hook? 

```tsx
import { useCallback } from 'react';

function useHandleBlurTextResource(
  language: string,
  updateTextResource: (args: MutationArgs) => void,
  doReloadPreview: () => void
) {
  return useCallback(
    (textResource: TextResource) => {
      const updatedTextResource = createTextResourceWithLanguage(language, textResource);
      const mutationArgs = convertTextResourceToMutationArgs(updatedTextResource);
      updateTextResource(mutationArgs);
      doReloadPreview();
    },
    [updateTextResource, doReloadPreview, language]
  );
}
```
It can then be used like this: 
```tsx
const handleBlurTextResource = useHandleBlurTextResource(language, updateTextResource, doReloadPreview);
```